src/corelibs/U2View/src/ov_msa/McaEditorNameList.cpp

```diff
@@ -126,4 +128,18 @@ int McaEditorNameList::getIconColumnWidth() const {
     return iconColumnWidth;
 }
 
+void McaEditorNameList::keyPressEvent(QKeyEvent *e) {
+    const int key = e->key();
+    const Qt::KeyboardModifiers modifiers = e->modifiers();
+    if (modifiers == Qt::NoModifier && (key == Qt::Key_Space || key == Qt::Key_Enter || key == Qt::Key_Return)) {
```

Comment:
There is a special function for this - `QAction::setShortcuts`. I think, it's better to use it.

Reply:
This is not what I need: the shortcut is active in the name-list only. 

When the focus is inside another element this shortcut does not work: keyboard navigation inside other MCA components is not defined today at all and is out of the scope of current PR.

Reply:
And I still dot see any contradictions. You may use `QAction::setShortcuts` and `QAction::setShortcutContext(Qt::WidgetShortcut)` to set the shortcut available for the certain widget only.

Reply:
There won't be a shortcut hint in context menu in this case. Nobody will know about the shortcut.
But I can't offer anything that would be definitely better.

Reply:
This can work but I see 2 problems:
1) The action is defined in the MCA editor globally, while the shortcut I want to have is valid only for MCAEditorList. It means that it is logically incorrect to define this shortcut globally.

2) I use 3 different keys to do the same: Space, Enter, Return. Can your solution handle all of them?

Reply:
1. Why not? If you want this shortcut for MCANameList only, it has to be created in MCANameList. Otherwise, the behavior is correct.
2. `QAction::setShortcuts(const QList<QKeySequence> &shortcuts)` allows you to set several shortcuts.

Reply:
1. This action is defined in MCA editor globally. Custom name-list shortcut is a customization of the action by the MCA name list widget. The action can be added to the context/tools menu, toolbar, activated using another user action (mouse click) and in this case, there is no need for a global shortcut.

2. Got it. Thanks

Reply:
I don't have more arguments about "1". I have presented my point of view, you may follow it or not, I won't insist.

Reply:
I think your suggestion is 100% valid but requires global shortcuts model design for MCA editor.

While this model is not ready any trivial keyboard shortcut like 'Space' or 'Enter' which has a high collision chance with other behaviors (like "edit sequence" where Space adds gaps) should be used in a very limited way.